packages/base/src/decorators/customElement.ts

```diff
@@ -1,16 +1,56 @@
+import type UI5Element from "../UI5Element.js";
+import type { TemplateFunction as Template } from "../renderer/executeTemplate.js";
+import type { ComponentStylesData as Styles } from "../types.js";
+import type { Renderer } from "../renderer/LitRenderer.js";
+
 /**
  * Returns a custom element class decorator.
  *
- * @param { string } tag
+ * @param { string | object } tagName
  * @returns { ClassDecorator }
  */
-const customElement = (tag: string): ClassDecorator => {
+const customElement = (tagName: string | {
+	tag?: string,
+	languageAware?: boolean,
+	themeAware?: boolean,
+	fastNavigation?: boolean,
+	renderer?: Renderer,
+	styles?: Styles,
+	template?: Template,
+	dependencies?: Array<typeof UI5Element>,
+	staticAreaStyles?: Styles,
+	staticAreaTemplate?: Template,
+}): ClassDecorator => {
 	return (target: any) => {
 		if (!Object.prototype.hasOwnProperty.call(target, "decoratorMetadata")) {
 			target.decoratorMetadata = {};
 		}
 
+		if (typeof tagName === "string") {
+			target.decoratorMetadata.tag = tagName;
+			return;
+		}
+
+		const {
+			tag,
+			languageAware = false,
+			themeAware = false,
+			fastNavigation = false,
+		 } = tagName;
+
 		target.decoratorMetadata.tag = tag;
+		target.decoratorMetadata.languageAware = languageAware;
+		target.decoratorMetadata.themeAware = themeAware;
+		target.decoratorMetadata.fastNavigation = fastNavigation;
+
+		["render", "renderer", "template", "staticAreaTemplate", "styles", "staticAreaStyles", "dependencies"].forEach((customElementEntity: string) => {
+			const _customElementEntity = customElementEntity === "render" ? "renderer" : customElementEntity;
+			const customElementEntityValue = tagName[_customElementEntity as keyof typeof tag];
```

Comment:
If they pass `render` and not `renderer`, won't `_customElementEntity` become `renderer`, and then we'll try to read this from `tagName`, and it won't exist?

Reply:
in Object.define 
we use `customElementEntity`
```js
Object.defineProperty(target, customElementEntity
```
so the "render" getter should be created, but I will check.

and _customElementEntity for reading the value that developers pass to "renderer" in the @ customElement decorator in the components.